pkg/docker/docker.go

```diff
@@ -151,3 +152,35 @@ func PushImage(ctx context.Context, log *zap.SugaredLogger, dryRun bool, image s
 
 	return nil
 }
+
+// Copy copies the content from a directory out of the
+// container onto the host system
+func Copy(ctx context.Context, log *zap.SugaredLogger, image string, dst string, src string) error {
+	var err error
+
+	log = log.With("image", image)
+
+	log.Infow("Extracting image...", "src", src, "dst", dst)
+
+	dst, err = filepath.Abs(dst)
+	if err != nil {
+		return fmt.Errorf("failed to determine absolute path: %v", err)
+	}
+
+	mountPoint := "/kubermaticextractor"
+	args := []string{
```

Comment:
sorry if I missed this in my previous pass, but here you should better use `docker cp` command instead. No need of running a container: 

https://docs.docker.com/engine/reference/commandline/cp/

Reply:
As stated in the PR description:

> Docker offers the docker cp command, but that needs a Docker container, so I chose to keep it simple and just use docker run. As the addons image should always be inherited from our image, this means that we always have Alpine available.

Also stated in the documentation:

> Copy files/folders between a **container** and the local filesystem

Reply:
So it's either `docker create + docker cp + docker delete`, or a single `docker run --rm` with much less state handling and need to cleanup if any of the steps failed.

Reply:
Yeah, you need to create the container, but you don't need to run it. I missed that sentence in the PR description. If we don't want to support pure data containers at the moment we can go for your solution as it is simpler.